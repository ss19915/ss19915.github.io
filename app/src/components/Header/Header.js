import React from 'react';
import { LightButton, DarkButton } from '../ThemeButtons';
import { THEME } from '../../constants';
import { setTheme } from '../../actions';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

const Header = (props) => {
    const {
        activeTheme,
        setTheme,
    } = props;

    return (
        <React.Fragment>
            <Link to='/' > Portfolio </Link>
            <Link to='/blogs' > Blogs </Link>
            <DarkButton
                disabled={activeTheme === THEME.DARK}
                onClick={() => setTheme(THEME.DARK)}
            >
                Dark Theme
            </DarkButton>
            <LightButton
                disabled={activeTheme === THEME.LIGHT}
                onClick={() => setTheme(THEME.LIGHT)}
            >
                Light Theme
            </LightButton>
        </React.Fragment>
    );
};


const mapDispatchToProps = (dispatch) => ({
    setTheme: (theme) => dispatch(setTheme(theme)),
});

const mapStateToProps = (state) => ({
    activeTheme: state.theme,
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);