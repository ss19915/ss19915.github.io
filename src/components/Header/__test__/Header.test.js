import Header from '../Header';
import { BrowserRouter } from 'react-router-dom';

describe('Component: Haeader', () => {
    it('Mount Snapshot Test', () => {
        const wrapper = mountWithThemeAndStore(<BrowserRouter><Header/></BrowserRouter>);

        expect(wrapper).toMatchSnapshot();
    });
});