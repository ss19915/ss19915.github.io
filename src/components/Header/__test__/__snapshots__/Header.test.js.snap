// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Haeader Mount Snapshot Test 1`] = `
.emotion-0 {
  padding: 5px;
  margin: 10px;
  border-radius: 5px;
  color: white;
  background-color: black;
}

.emotion-0:disabled:hover {
  cursor: none;
}

.emotion-0:hover {
  cursor: pointer;
}

.emotion-2 {
  padding: 5px;
  margin: 10px;
  border-radius: 5px;
  color: black;
  background-color: white;
}

.emotion-2:disabled:hover {
  cursor: none;
}

.emotion-2:hover {
  cursor: pointer;
}

<ThemeProvider
  theme={
    Object {
      "cursor_disabled": "none",
      "cursor_hover": "pointer",
      "height_max": "100vh",
      "primaryBackgroundColor": "black",
      "primaryColor": "white",
      "spacing_l": "10px",
      "spacing_m": "5px",
      "spacing_s": "2px",
      "spacing_xl": "15px",
      "width_max": "100%",
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <BrowserRouter>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(Header)>
          <Header
            setTheme={[Function]}
          >
            <Link
              to="/"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                 Portfolio 
              </a>
            </Link>
            <Link
              to="/blogs"
            >
              <a
                href="/blogs"
                onClick={[Function]}
              >
                 Blogs 
              </a>
            </Link>
            <DarkButton
              disabled={false}
              onClick={[Function]}
            >
              <button
                className="emotion-0 emotion-1"
                disabled={false}
                onClick={[Function]}
              >
                Dark Theme
              </button>
            </DarkButton>
            <LightButton
              disabled={false}
              onClick={[Function]}
            >
              <button
                className="emotion-2 emotion-3"
                disabled={false}
                onClick={[Function]}
              >
                Light Theme
              </button>
            </LightButton>
          </Header>
        </Connect(Header)>
      </Router>
    </BrowserRouter>
  </Provider>
</ThemeProvider>
`;
